test/validation/api/thread/thread.c

```diff
@@ -104,13 +104,27 @@ static void thread_test_odp_thread_count(void)
 {
 	int count = odp_thread_count();
 
-	/* One thread running */
+	/* One control thread running */
 	CU_ASSERT(count == 1);
+	CU_ASSERT(odp_thread_control_count() == 1);
+	CU_ASSERT(odp_thread_control_count() <= odp_thread_control_count_max());
+	CU_ASSERT(odp_thread_worker_count() == 0);
 
 	CU_ASSERT(count >= 1);
 	CU_ASSERT(count <= odp_thread_count_max());
 	CU_ASSERT(count <= ODP_THREAD_COUNT_MAX);
+}
+
+static void thread_test_odp_thread_count_max(void)
+{
+	CU_ASSERT(odp_thread_count_max() > 0);
 	CU_ASSERT(odp_thread_count_max() <= ODP_THREAD_COUNT_MAX);
+
+	CU_ASSERT(odp_thread_control_count_max() >= 0);
+	CU_ASSERT(odp_thread_control_count_max() <= odp_thread_count_max());
+
+	CU_ASSERT(odp_thread_worker_count_max() >= 0);
+	CU_ASSERT(odp_thread_worker_count_max() <= odp_thread_count_max());
```

Comment:
Are the max amounts expected to change dynamically? If not, they could be saved to variables once and then compare those.

Reply:
The values are not expected to change, so modified the test in V2.